app/models/manageiq/providers/base_manager/refresher.rb

```diff
@@ -138,6 +140,22 @@ def save_inventory(ems, _target, persister)
         InventoryRefresh::SaveInventory.save_inventory(ems, persister.inventory_collections)
       end
 
+      def publish_inventory(ems, target, persister)
+        return if MiqQueue.messaging_type == "miq_queue"
+
+        log_header = format_ems_for_logging(ems)
+        target_identifier = "#{target.class}__#{target.name}__#{target.id}"
+
+        MiqQueue.messaging_client('event_handler')&.publish_topic(
+          :service => "manageiq.ems-inventory",
+          :sender  => ems.id,
+          :event   => target_identifier,
+          :payload => persister.to_json
```

Comment:
There are limits to the payload size in kafka I think, so do we have to be careful wit that here?

Reply:
Yes, kafka is optimized for _lots_ of small messages so I'm thinking that one message per inventory item with the context of if it was created/updated/deleted will work better.